docs/secure-coding-guidelines.md

```diff
@@ -0,0 +1,172 @@
+
+#### Passwords
+
+- Passwords used to decrypt content should adhere to security team recommendations
+
+### Logging & Error Handling
+
+#### Logging
+
+- Secrets and **sensitive** data should not be logged
+
+#### Error Handling
+
+- Avoid including data directly in error messages, especially **sensitive** data
+
+  For example, the error "Invalid hex data: \_\_\_\_" might indadvertantly leak a private key. Instead the error could be made more generic ("Invalid hex data"), or you could describe the problem without embedding the data ("Invalid hex data; missing 0x prefix").
+
+### Third Party Integrations
+
+#### Authentication and Authorization
+
+- Do not bundle API tokens in client side applications unless you intend for them to be publicly accessible
+
+#### Integration/Application Integrity
+
+- Monitor integrations/applications for changes and security issues
+
+### Third Party Applications
+
+#### Permissions
+
+- All applications/dapps/snaps shall adhere to the following to Principle of Least Privilege with a default state of allowing no permissions
+- Users shall provide consent for required permissions of applications/dapps/snaps
+
+### Dependency Management
+
+#### Dependency Integrity
+
+- Use a lockfile or pinned dependencies to maintain control over which version of each dependency is used
+
+#### Avoid Deprecated and Unmaintained Packages
+
+- Using Socket.dev to check the health and maintenance status of dependencies and seek alternatives when necessary
+
+#### Reduce Dependencies and Keep Them Minimal
+
+- Minimize dependencies to reduce potential vulnerabilities
+
+#### Audit and Monitor Dependencies
+
+- Monitor dependencies for security vulnerabilities and other problems
+  - Periodically scan for security vulnerabilities (e.g. using tools like `npm audit`)
+  - Update dependencies quickly when they have security vulnerabilities
+  - Use Socket.dev to monitor dependencies for other noteworthy changes, such as maintainer changes, or the addition of install scripts or binary files
+    - Use the following etiquette when addressing Socket.dev warnings:
+      - Investigate and address all warnings before merging a PR
+      - Avoid using the `ignore-all` bot command, instead ignoring each warning one at a time
+      - If you've investigated a warning and found that it's not indicative of a malicious dependency, ignore it with a bot comment and explain your investigation with a short comment
+      - Contact the security team if you're unsure how to investigate something, or if you'd like to disable a warning category
+
+#### LavaMoat (JavaScript projects only)
+
+- LavaMoat `allow-scripts` should be enabled on all projects
+  - This project relies upon install scripts being disabled in your package manager. This can be verified by adding the dependency `@lavamoat/preinstall-always-fail`, which will cause installation to fail if scripts are enabled.
```

Comment:
`allow-scripts setup` adds that dependency. Also, yarn 3 and 4 will not fail the entire install process because they swallow errors.

Reply:
Ah I see, I hadn't realized that about Yarn v3/v4. I guess your point here is that it'd be safer to advise using `allow-scripts setup` because users might fail to properly disable install scripts in the package manager otherwise, is that correct? Just wanted to confirm that setting `ignore-scripts` is also something handled by the setup script.